testing/unittests/metrics/MetricFrameworkTests.cpp

```diff
@@ -209,13 +210,62 @@ class MetricFrameworkTests : public CppUnit::TestFixture
             ScopedGaugeDecrementer gaugeDecrementer(*pGauge, 1);
         }
         CPPUNIT_ASSERT_EQUAL(0, static_cast<int>(pGauge->queryValue()));
+    }
+
+    void Test_metric_meta_data()
+    {
+        MetricMetaData data1{MetricMetaDataItem{"key1", "value1"}};
```

Comment:
minor: the keyword MetricMetaDataItem can be removed.

`MetricMetaData data1{{"key1", "value1"}};`

main reason for mentioning it is that this code is likely to serve as an example of how it should be called.

You can also include it directly on the call:

`std::make_shared<CounterMetric>("requests.completed", "description", SMeasureCount, MetricMetaData{{"key1", "value1"}})
`

but for some reason the MetricMetaData cannot be removed (even though it can be removed from a call to the constructor.  Looks like an oddity of template parameter forwarding.